### PR TITLE
Extend the DatadogAgent CR example

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

Add the common config that can be used in the Datadog Agent CR to enable agent features

### Motivation

Better documentation

### Additional Notes

This example CR will be used as example for the operator on `OperatorHub`

### Describe your test plan

Edit the CR, enable everything, deploy it and make sure it works as expected 💯

### CODE REVIEW

File: config/samples/datadog-agent-all.yaml [hunk 1]
```diff
@@ -3,19 +3,41 @@ kind: DatadogAgent
...
   name: datadog
 spec:
+  clusterName: <CLUSTER_NAME>
   credentials:
     apiKey: <DATADOG_API_KEY>
     appKey: <DATADOG_APP_KEY>
+    token: <DATADOG_CLUSTER_AGENT_TOKEN>
   agent:
     image:
       name: "datadog/agent:latest"
```
Comment (vboulineau) [inline config/samples/datadog-agent-all.yaml:13]: Not sure we should include `latest` here, we should be in line with Helm chart and use fixed version. In terms of example we should maybe remove this option to let the operator choose

Comment (ahmed-mez) [reply config/samples/datadog-agent-all.yaml:13]: currently the operator doesn't have that logic yet (to default image.name). I guess we'll have to put a version here

File: config/samples/datadog-operator-hub-example.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  name: datadog
+spec:
+  clusterName: <CLUSTER_NAME>
+  credentials:
+    apiKey: <DATADOG_API_KEY>
+    appKey: <DATADOG_APP_KEY>
+    token: <DATADOG_CLUSTER_AGENT_TOKEN>
+  agent:
+    image:
+      name: "datadog/agent:7.24.1"
```
Comment (L3n41c) [inline config/samples/datadog-operator-hub-example.yaml:13]: Should we switch to the gcr images instead ?
```suggestion
      name: "gcr.io/datadoghq/agent:7.24.1"
```

File: config/samples/datadog-operator-hub-example.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      bpfDebugEnabled: false
+    security:
+      compliance:
+        enabled: false
+      runtime:
+        enabled: false
+  clusterAgent:
+    replicas: 2
+    image:
+      name: "datadog/cluster-agent:1.10.0"
```
Comment (L3n41c) [inline config/samples/datadog-operator-hub-example.yaml:33]: Should we switch to the gcr images instead ?
```suggestion
      name: "gcr.io/datadoghq/cluster-agent:1.10.0"
```

**Review by L3n41c — APPROVED** [review]

**Pull request merged**